ctapipe_io_lst/calibration.py

```diff
@@ -163,6 +163,7 @@ def __init__(self, subarray, config=None, parent=None, **kwargs):
         self.last_readout_time = {}
         self.first_cap = {}
         self.first_cap_old = {}
+        self.spike_flag = {}
```

Comment:
this is only used to tag the changed slices in the code, it is not used for actual calculation, right?

have you checked if the subtraction code and the old interpolation code are similar in speed?
removing those spike_flag tags could speed it up a bit more.